### ISSUE TITLE
Very slow training with Qwen Edit 2509

### ISSUE THREAD

**Issue by gianlucaman** [kohya-ss/musubi-tuner#713]
Hello, I tried to run a training last night and this morning it was still at 8% for the first step. It was incredibly slow so I guess I did something wrong. I have a 3090 ti.

This is the script I'm sending through cmd:

accelerate launch --num_cpu_threads_per_process 1 --mixed_precision bf16 ^
src\musubi_tuner\qwen_image_train_network.py ^
--dit "D:\musubi-tuner\models\qwen_image_edit_2509_bf16.safetensors" ^
--vae "D:\musubi-tuner\models\vae.safetensors" ^
--text_encoder "D:\musubi-tuner\models\qwen_2.5_vl_7b.safetensors" ^
--dataset_config "D:\musubi-tuner\trainings\borsa_iblues_rossa\train_qwen_image.toml" ^
--edit_plus --sdpa --mixed_precision bf16 ^
--timestep_sampling shift --weighting_scheme none --discrete_flow_shift 2.2 ^
--optimizer_type AdamW --learning_rate 1e-5 --gradient_checkpointing ^
--max_data_loader_n_workers 2 --persistent_data_loader_workers ^
--network_module networks.lora_qwen_image --network_dim 16 ^
--max_train_epochs 10 --save_every_n_epochs 1 --seed 42 ^
--output_dir "D:\musubi-tuner\trainings\borsa_iblues_rossa\output" ^
--output_name "borsa_iblues_rossa_lora"
--use_pinned_memory_for_block_swap

This is my training.toml:

[general]
resolution = [1328, 1328] # optional, [W, H], default is [960, 544]. This is the default resolution for all datasets
caption_extension = ".txt" # optional, default is None. This is the default caption extension for all datasets
batch_size = 1 # optional, default is 1. This is the default batch size for all datasets
num_repeats = 1 # optional, default is 1. Number of times to repeat the dataset. Useful to balance the multiple datasets with different sizes.
enable_bucket = true # optional, default is false. Enable bucketing for datasets
bucket_no_upscale = false # optional, default is false. Disable upscaling for bucketing. Ignored if enable_bucket is false

[[datasets]]
image_directory = "D:/musubi-tuner/trainings/borsa_iblues_rossa/dataset/images"
cache_directory = "D:/musubi-tuner/trainings/borsa_iblues_rossa/dataset/cache" # optional, default is None to use the same directory as the image directory. NOTE: caching is always enabled
control_directory = "D:/musubi-tuner/trainings/borsa_iblues_rossa/dataset/control" # optional, required for dataset with control images

Can you help me suggesting some more adapt scripts to train in a good amount of time?

Thank you so much in advance. I can give more infos if needed.

**Comment by kohya-ss** [comment]
Please add `--fp8_base --fp8_scaled` options for fp8 training, and `--blocks_to_swap 30` to enable block swap. If dedicated VRAM is not enough, increase the number of block swap. If you have free dedicated VRAM, reducing it from 30 will make the training faster.

**Comment by gianlucaman** [comment]
Thank you so much. I will try with the new script. 

By the way the terminal keeps telling me that it's failing importing sageattention, as I didn't install it. Does it make a huge difference in terms of speed?
